From 159bbb6708538a0edc473ac969d54efd7234e907 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?M=C4=81rti=C5=86=C5=A1=20Ma=C4=8Ds?=
 <macs.martins@gmail.com>
Date: Tue, 27 Sep 2016 12:22:06 +0300
Subject: [PATCH 2/5] Scroll by half-pages

---
 surf.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/surf.c b/surf.c
index 20b9fd8..831ed2f 100644
--- a/surf.c
+++ b/surf.c
@@ -1297,10 +1297,8 @@ scroll(GtkAdjustment *a, const Arg *arg)
 	switch (arg->i) {
 	case +10000:
 	case -10000:
-		v += gtk_adjustment_get_page_increment(a) * (arg->i / 10000);
+		v += gtk_adjustment_get_page_increment(a) * (arg->i / 10000) / 2.0;
 		break;
-	case +20000:
-	case -20000:
 	default:
 		v += gtk_adjustment_get_step_increment(a) * arg->i;
 	}
-- 
2.7.3

