From 7e2f3fca5580567989d4c33f34c41f2eaf2feafa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?M=C4=81rti=C5=86=C5=A1=20Ma=C4=8Ds?=
 <macs.martins@gmail.com>
Date: Sun, 9 Oct 2016 02:45:10 +0300
Subject: [PATCH 3/3] Reset cookie policy when creating new window

---
 surf.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/surf.c b/surf.c
index 831ed2f..953a571 100644
--- a/surf.c
+++ b/surf.c
@@ -546,6 +546,8 @@ WebKitWebView *
 createwindow(WebKitWebView  *v, WebKitWebFrame *f, Client *c)
 {
 	Client *n = newclient();
+	policysel = strlen(cookiepolicies);
+	togglecookiepolicy(c, NULL);
 	return n->view;
 }
 
-- 
2.7.3

