--- zathura-0.3.7-orig/zathura/callbacks.c	2016-03-08 22:48:07.000000000 +0200
+++ zathura-0.3.7/zathura/callbacks.c	2017-01-29 06:39:10.911603503 +0200
@@ -323,14 +323,14 @@
   unsigned int number_of_pages = zathura_document_get_number_of_pages(zathura->document);
   for (unsigned int page_id = 0; page_id < number_of_pages; page_id++) {
     zathura_page_t* page = zathura_document_get_page(zathura->document, page_id);
-    if (page == NULL || zathura_page_get_visibility(page) == false) {
+    if (page == NULL) {
       continue;
     }
 
     GtkWidget* page_widget = zathura_page_get_widget(zathura, page);
     g_object_set(G_OBJECT(page_widget), "draw-links", FALSE, NULL);
 
-    if (eval == true) {
+    if (eval == true && zathura_page_get_visibility(page) == true) {
       zathura_link_t* link = zathura_page_widget_link_get(ZATHURA_PAGE(page_widget), index);
 
       if (link != NULL) {
