--- zathura-0.3.7-orig/zathura/completion.c	2017-01-11 23:05:00.000000000 +0200
+++ zathura-0.3.7/zathura/completion.c	2017-01-29 17:34:55.110700851 +0200
@@ -198,7 +198,7 @@
     if (girara_list_size(recent_files) != 0) {
       const size_t path_len = strlen(path);
       GIRARA_LIST_FOREACH(recent_files, const char*, iter, file)
-        if (strncmp(path, file, path_len) == 0) {
+        if (*input == '\0' || strncmp(path, file, path_len) == 0) {
           girara_debug("adding %s (recent file)", file);
           girara_completion_group_add_element(history_group, file, NULL);
         }
