From ceba4e75a31ab028b10799c656cc3244f1309e20 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?M=C4=81rti=C5=86=C5=A1=20Ma=C4=8Ds?=
 <macs.martins@gmail.com>
Date: Sat, 8 Apr 2017 11:26:21 +0300
Subject: [PATCH 4/7] urgency hint

---
 surf.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/surf.c b/surf.c
index 74ced43..3c3d20b 100644
--- a/surf.c
+++ b/surf.c
@@ -1104,6 +1104,11 @@ winevent(GtkWidget *w, GdkEvent *e, Client *c)
 	int i;
 
 	switch (e->type) {
+	case GDK_FOCUS_CHANGE:
+		if (e->focus_change.in) {
+			gtk_window_set_urgency_hint(GTK_WINDOW(c->win), FALSE);
+		}
+		break;
 	case GDK_ENTER_NOTIFY:
 		c->overtitle = c->targeturi;
 		updatetitle(c);
@@ -1214,6 +1219,8 @@ createwindow(Client *c)
 	                 G_CALLBACK(winevent), c);
 	g_signal_connect(G_OBJECT(w), "window-state-event",
 	                 G_CALLBACK(winevent), c);
+	g_signal_connect(G_OBJECT(w), "focus-in-event",
+	                 G_CALLBACK(winevent), c);
 
 	return w;
 }
@@ -1249,6 +1256,7 @@ loadchanged(WebKitWebView *v, WebKitLoadEvent e, Client *c)
 		    enablescrollbars ? "auto" : "hidden");
 		*/
 		runscript(c);
+		gtk_window_set_urgency_hint(GTK_WINDOW(c->win), TRUE);
 		break;
 	}
 	updatetitle(c);
-- 
2.10.2

