From aee5a2dae0bd103f1657deda231e19bbf5f618c2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?M=C4=81rti=C5=86=C5=A1=20Ma=C4=8Ds?=
 <macs.martins@gmail.com>
Date: Sat, 8 Apr 2017 09:22:48 +0300
Subject: [PATCH 8/8] Better scrolling

---
 surf.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/surf.c b/surf.c
index 3cf9642..e650125 100644
--- a/surf.c
+++ b/surf.c
@@ -1642,6 +1642,12 @@ zoom(Client *c, const Arg *a)
 void
 scroll(Client *c, const Arg *a)
 {
+	if (strchr("UD", a->i)) {
+		evalscript(c, JSCNone, "window.scrollBy(0, %f * window.innerHeight)",
+		           (a->i == 'U' ? -1 : 1) * 0.42);
+		return;
+	}
+
 	GdkEvent *ev = gdk_event_new(GDK_KEY_PRESS);
 
 	gdk_event_set_device(ev, gdkkb);
@@ -1650,24 +1656,18 @@ scroll(Client *c, const Arg *a)
 	ev->key.time = GDK_CURRENT_TIME;
 
 	switch (a->i) {
+	case 'u':
+		ev->key.keyval = GDK_KEY_Up;
+		break;
 	case 'd':
 		ev->key.keyval = GDK_KEY_Down;
 		break;
-	case 'D':
-		ev->key.keyval = GDK_KEY_Page_Down;
-		break;
 	case 'l':
 		ev->key.keyval = GDK_KEY_Left;
 		break;
 	case 'r':
 		ev->key.keyval = GDK_KEY_Right;
 		break;
-	case 'U':
-		ev->key.keyval = GDK_KEY_Page_Up;
-		break;
-	case 'u':
-		ev->key.keyval = GDK_KEY_Up;
-		break;
 	}
 
 	gdk_event_put(ev);
-- 
2.10.2

