From 4b7d1f722c58c9986e9bea9124f2087a1f12ce79 Mon Sep 17 00:00:00 2001
From: Sascha Kruse <knopwob@googlemail.com>
Date: Wed, 27 Aug 2014 09:44:31 +0200
Subject: [PATCH] fix '-' and '_' matching in urls

fix issue #192
---
 menu.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/menu.c b/menu.c
index 9105b48..d17cc9b 100644
--- a/menu.c
+++ b/menu.c
@@ -29,8 +29,8 @@ char *extract_urls(const char *to_match)
         if (!is_initialized) {
                 char *regex =
                     "\\b(https?://|ftps?://|news://|mailto:|file://|www\\.)"
-                    "[[:alnum:]\\-\\@;/?:&=%$.+!*\x27,~#]*"
-                    "(\\([[:alnum:]\\-\\@;/?:&=%$.+!*\x27,~#]*\\)|[[:alnum:]\\-\\@;/?:&=%$+*~])+";
+                    "[-[:alnum:]_\\@;/?:&=%$.+!*\x27,~#]*"
+                    "(\\([-[:alnum:]_\\@;/?:&=%$.+!*\x27,~#]*\\)|[-[:alnum:]_\\@;/?:&=%$+*~])+";
                 int ret = regcomp(&cregex, regex, REG_EXTENDED | REG_ICASE);
                 if (ret != 0) {
                         printf("failed to compile regex\n");
-- 
2.7.3

