--- a/alot/db/utils.py
+++ b/alot/db/utils.py
@@ -422,7 +422,7 @@ def remove_cte(part, as_string=False):
         # we need to encode that back to bytes so we can decode it using
         # the correct encoding, or it might not, in which case assume that
         # the str representation we got is correct.
-        bp = payload.encode('raw-unicode-escape')
+        bp = payload.encode('utf-8')
 
     elif 'quoted-printable' in cte:
         logging.debug('assuming Content-Transfer-Encoding: quoted-printable')
-- 
2.21.0

