--- a/alot/widgets/thread.py
+++ b/alot/widgets/thread.py
@@ -53,6 +53,14 @@ class MessageSummaryWidget(urwid.WidgetWrap):
                 cols.append(('fixed', tag_widget.width(), tag_widget))
         line = urwid.AttrMap(urwid.Columns(cols, dividechars=1), attr,
                              focus_att)
+
+        def _render_wrap(size, focus=False):
+            for c in cols:
+                if isinstance(c, tuple):
+                    c[2].set_map('focus' if focus else 'normal')
+            return urwid.AttrMap.render(line, size, focus)
+        line.render = _render_wrap
+
         urwid.WidgetWrap.__init__(self, line)
 
     def __str__(self):
-- 
2.21.0

