--- mupdf-1.14.0-source-orig/source/html/html-layout.c	2016-11-28 15:34:04.000000000 +0200
+++ mupdf-1.14.0-source/source/html/html-layout.c	2017-01-27 18:12:47.577100475 +0200
@@ -2072,7 +2072,6 @@
 	fz_rect bbox;
 	const char *dest;
 	const char *href;
-	float end;
 
 	while (flow)
 	{
@@ -2080,27 +2079,27 @@
 		next = flow->next;
 		if (href && (int)(flow->y / page_h) == page)
 		{
-			/* Coalesce contiguous flow boxes into one link node */
-			end = flow->x + flow->w;
-			while (next &&
-					next->y == flow->y &&
-					next->h == flow->h &&
-					has_same_href(next->box, href))
-			{
-				end = next->x + next->w;
-				next = next->next;
-			}
+#define CALC_BOX(rect, html_flow) \
+	rect.x0 = html_flow->x; \
+	rect.y0 = html_flow->y - page * page_h - (html_flow->type != FLOW_IMAGE ? 0.76 * html_flow->h : 0); \
+	rect.x1 = html_flow->x + html_flow->w; \
+	rect.y1 = rect.y0 + html_flow->h;
+
+			CALC_BOX(bbox, flow)
 
-			bbox.x0 = flow->x;
-			bbox.y0 = flow->y - page * page_h;
-			bbox.x1 = end;
-			bbox.y1 = bbox.y0 + flow->h;
-			if (flow->type != FLOW_IMAGE)
+			/* Coalesce contiguous/overlapping flow boxes into one link node */
+			while (next && has_same_href(next->box, href))
 			{
-				/* flow->y is the baseline, adjust bbox appropriately */
-				bbox.y0 -= 0.8f * flow->h;
-				bbox.y1 -= 0.8f * flow->h;
+				fz_rect bbox_next;
+				CALC_BOX(bbox_next, next)
+
+				if (fz_abs((bbox.x0 + bbox.x1) - (bbox_next.x0 + bbox_next.x1)) - (bbox.x1 - bbox.x0 + bbox_next.x1 - bbox_next.x0) > 0.0001 ||
+				    fz_abs((bbox.y0 + bbox.y1) - (bbox_next.y0 + bbox_next.y1)) - (bbox.y1 - bbox.y0 + bbox_next.y1 - bbox_next.y0) > 0.0001) break;
+
+				bbox = fz_union_rect(bbox, bbox_next);
+				next = next->next;
 			}
+#undef CALC_BOX
 
 			if (is_internal_uri(href))
 			{
